pkgs/applications/graphics/apitrace/default.nix

```diff
@@ -15,7 +15,31 @@ stdenv.mkDerivation rec {
   # of games -- so it's fine to use e.g. bundled snappy.
   buildInputs = [ libX11 procps python2 libdwarf qtbase qtwebkit ];
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
+  # Don't automatically wrap all binaries, I prefer to explicitly only wrap
+  # `qapitrace`.
+  dontWrapQtApps = true;
+
+  postFixup = ''
+    # Apitrace wraps some program. For some reason, they are not able to find
+    # the OpenGL driver anymore.
+    # We add back the position of the driver to the library search path.
+    # Adding the path for path 32 and 64 bits have no impact on the closure and
+    # is simpler because at build time of api trace, we don't know if the
+    # traced program will be 64 or 32 bits.
+    wrapProgram $out/bin/apitrace \
+      --prefix LD_LIBRARY_PATH : "/run/opengl-driver/lib:/run/opengl-driver-32/lib"
```

Comment:
Is that not inherited from the parent environment? I think these paths are NixOS-specific.

Reply:
`LD_LIBRARY_PATH` points to these on NixOS:

https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/hardware/opengl.nix#L150-L151

I don't see a reason to wrap with these paths on non-NixOS.

Reply:
Well, on non-nixos these directory does not exists (and that's the reason why OpenGL does not work on non nixos by default).
This is not inherited now. This changed recently (I'm too lazy to find the commit reference). But now it appear in the runpath of all the OpenGL binaries in nix. Search for `addGLRunpath` in the nixpkgs. However, for reasons that I don't understand, once wrapped by `apitrace`, binaries are not able to find the opengl lib anymore.

This fix is dubious, but I cannot find a better solution.

Reply:
Hmm - there has been a bit of back and forth on having these OpenGL paths in `LD_LIBRARY_PATH` vs. `RUNPATH`, but once opengl is enabled, these global variables should be set.

I can't see how this wrapper script should change the behaviour, if it's also set on your global system. Can you confirm if this is the case on your system?

Reply:
Or what do you invoke when using this package? Is it `qapitrace`, which might clear `LD_LIBRARY_PATH`?

Reply:
I'm invoking `apitrace`. Neither `LD_LIBRARY_PATH` nor `RUNPATH` are set on a nixos-unstable system. There is nothing related to `opengl` in the `env`.

The current situation is that for some OpenGL binaries are wrapped using `addOpenGLRunpath`. But for reasons that I don't understand, `apitrace` breaks this.

I'll do more investigation.

Reply:
Could it be you didn't do `hardare.opengl.enable = true` in your configuration? My [earlier sent snippet](https://github.com/NixOS/nixpkgs/pull/86475#discussion_r418722639) only gets applied when you enable it, but assuming you never rebooted since then, there might be dangling symlinks in `/run/opengl-driver*`…

Most desktop manager modules do enable this, but you might have a custom configuration.

Reply:
@flokli If I understand your snippet correctly, it depends on: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/hardware/opengl.nix#L111-L121 which is `false` by default, so it may be the reason why I don't have the `LD_LIBRARY_PATH` in my system.

Howewer, you are right, I had no `hardware.opengl.enable = true` setting in my configuration. Setting it does not change anything however.

Reply:
There is no more wrapper for `LD_LIBRARY_PATH`. I'm overriding the `rpath` with the `/run/opengl-driver` path. I don't really understand why this is needed, but considering that:
- It fixs it
- It does not increase the runtime closure
- I don't see how it can break anything.

I do think not fully understanding the problem is fine ;)